### ISSUE TITLE
Building v0.25.3 fails at MacOS

### ISSUE THREAD

**Issue by ink-splatters** [gimli-rs/object#330]
## How to Repoduce
```Shell
bash-5.1$ cat Cargo.toml 
[package]
name = "object-test"
version = "0.1.0"
edition = "2018"
[dependencies]
object  = "0.25.3"

bash-5.1$ cargo build --release
```

## Build Log
```Shell
   Compiling object v0.25.3
error[E0583]: file not found for module `read_cache`
  --> /Users/inksplatters/.cargo/registry/src/github.com-1ecc6299db9ec823/object-0.25.3/src/read/mod.rs:14:1
   |
14 | mod read_cache;
   | ^^^^^^^^^^^^^^^
   |
   = help: to create the module `read_cache`, create file "/Users/inksplatters/.cargo/registry/src/github.com-1ecc6299db9ec823/object-0.25.3/src/read/read_cache.rs"

error[E0583]: file not found for module `dyld_cache`
 --> /Users/inksplatters/.cargo/registry/src/github.com-1ecc6299db9ec823/object-0.25.3/src/read/macho/mod.rs:8:1
  |
8 | mod dyld_cache;
  | ^^^^^^^^^^^^^^^
  |
  = help: to create the module `dyld_cache`, create file "/Users/inksplatters/.cargo/registry/src/github.com-1ecc6299db9ec823/object-0.25.3/src/read/macho/dyld_cache.rs"

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0583`.
error: could not compile `object`

To learn more, run the command again with --verbose.
```

## Workaround?

Funny, but I followed `rustc` advice to create the files

```Shell
bash-5.1$ touch /Users/inksplatters/.cargo/registry/src/github.com-1ecc6299db9ec823/object-0.25.3/src/read/mod.rs
bash-5.1$ touch /Users/inksplatters/.cargo/registry/src/github.com-1ecc6299db9ec823/object-0.25.3/src/read/macho/mod.rs
```

and I've got the compilation succeeded:

```Shell
bash-5.1$ cargo build --release                                                                                               
   Compiling object v0.25.3
   Compiling object-test v0.1.0 (/private/tmp/object-test)
    Finished release [optimized] target(s) in 3.06s
```

### OS

```Shell
bash-5.1$ sw_vers
ProductName:	macOS
ProductVersion:	11.4
BuildVersion:	20F71
```

### rustc

```Shell
bash-5.1$ rustc -vV
rustc 1.53.0 (53cb7b09b 2021-06-17)
binary: rustc
commit-hash: 53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b
commit-date: 2021-06-17
host: x86_64-apple-darwin
release: 1.53.0
LLVM version: 12.0.1
```

**Comment by mstange** [comment]
According to the source view on docs.rs, `src/read/read_cache.rs` is included in the published crate for 0.25.3:
https://docs.rs/crate/object/0.25.3/source/src/read/read_cache.rs

**Comment by philipc** [comment]
I don't have macOS some unable test your reproduction there, but it isn't a problem on Windows.

It's not a problem with the published crate. You can verify it contains these files with:
```
curl https://crates.io/api/v1/crates/object/0.25.3/download -L --output object.tar.gz
tar tzf object.tar.gz
```

This crate is used widely enough that people would have been complaining about it already if this was broken.

So I think your local registry is corrupted somehow. You could try deleting `~/.cargo/registry`, but I can't provide more help than that.
